mne/time_frequency/tfr.py

```diff
@@ -2464,23 +2533,23 @@ def plot_topo(
         info, data = _prepare_picks(info, data, picks, axis=0)
         del picks
 
-        # TODO this is the only remaining call to _preproc_tfr; should be refactored
-        #      (to use _prep_data_for_plot?)
-        data, times, freqs, vmin, vmax = _preproc_tfr(
+        # baseline, crop, convert complex to power, aggregate tapers, and dB scaling
+        data, times, freqs = _prep_data_for_plot(
             data,
             times,
             freqs,
-            tmin,
-            tmax,
-            fmin,
-            fmax,
-            mode,
-            baseline,
-            vmin,
-            vmax,
-            dB,
-            info["sfreq"],
+            tmin=tmin,
+            tmax=tmax,
+            fmin=fmin,
+            fmax=fmax,
+            baseline=baseline,
+            mode=mode,
+            dB=dB,
+            taper_weights=self.weights,
+            verbose=verbose,
         )
+        # get vlims
+        vmin, vmax = _setup_vmin_vmax(data, vmin, vmax)
```

Comment:
Seemed like as good as time as any to refactor and replace the `_preproc_tfr` call with `_prep_data_for_plot` where changes for handling data with a taper dimension have been made.